src/connector/src/source/nexmark/source/reader.rs

```diff
@@ -116,36 +102,26 @@ impl NexmarkSplitReader {
     #[try_stream(boxed, ok = Vec<SourceMessage>, error = anyhow::Error)]
     async fn into_stream(mut self) {
         let mut last_event = None;
+        let wall_clock_base_time = Instant::now();
         loop {
             let mut msgs: Vec<SourceMessage> = vec![];
             let old_events_so_far = self.generator.events_so_far();
 
-            // Get unix timestamp in milliseconds
-            let current_timestamp_ms = if self.use_real_time {
-                SystemTime::now()
-                    .duration_since(UNIX_EPOCH)
-                    .unwrap()
-                    .as_millis() as u64
-            } else {
-                0
-            };
-
             if let Some(event) = last_event.take() {
                 msgs.push(event);
             }
 
             let mut finished = false;
 
             while (msgs.len() as u64) < self.max_chunk_size {
-                if self.event_num > 0 && self.generator.events_so_far() >= self.event_num as u64 {
+                let event = self.generator.next().unwrap();
+
+                if self.event_num > 0 && self.generator.events_so_far() > self.event_num as u64 {
                     finished = true;
                     break;
                 }
-                let event = self.generator.next().unwrap();
 
-                if event.event_type() != self.event_type
-                    || self.generator.events_so_far() % self.split_num as u64
-                        != self.split_index as u64
+                if self.generator.events_so_far() % self.split_num as u64 != self.split_index as u64
```

Comment:
Is it even possible to avoid generating the events that are not in the current split (index)?

Reply:
Yes. But it seems a little invasive to put it into the generator. 🤔

Reply:
So I guess the throughput of nexmark generator cannot be improved with multi splits on multi cores? 🤣

Reply:
Of course this needs to be optimized. I'm only thinking of a elegant way...

Reply:
I found a bug #6450 that some splits may have no events at all. The whole split partitioning might need a reconsideration. 🥵

Reply:
🥵🥵🥵🥵